### ISSUE TITLE
Internal link syntax

### ISSUE THREAD

**Issue by DutchPete** [Zettlr/Zettlr#320]
## Description
I have a file with the name AFR_SubSah 20190910113250.md
In another file I have a link [[AFR_SubSah 20190910113250.md]]
The syntax consists of 2 parts, and for <kbd>CTRL</kbd> + click to work I have to click on numeric part of the link.
But when the link only consists of 20190910113250 or 20190910113250.md, it does not work either.
Furthermore, if I have a file with only a textual name and no numeric part, the link does not work either.
So, how does this work?

On your [page](https://docs.zettlr.com/en/academic/zkn-method/) it says:
"The second way that such a link may yield an exact match would be if the link's contents contain an ID in the format `[[<your-id>]]`. If any file has the ID `<your-id>`, Zettlr will also yield an exact match."

## Expected behaviour

**Screenshots**

## Platform
 - OS: Linux Mint
 - OS Version 19.2
 - Zettlr Version 1.4.1
 - Screen Resolution [only necessary if the bug is relating to the GUI]

**Comment by Shandi97** [comment]
If I may chime in. Not sure if I understand it correctly. I tried your link myself and it works.
Also isn't the file extension redundant?
![Imgur](https://i.imgur.com/d6OmbG9.gif)
I also use a good number of plain name links outside of ZK and they work.
![Imgur](https://i.imgur.com/qiDjaj3.gif)

**Comment by nathanlesage** [comment]
Please check these things for me:

Is the ID which is positioned within your filename detected as the ID of your file? (To see it, go to thin or expanded mode and display file meta, shortcut `Ctrl+Alt+S`)

The files obviously have to be loaded in Zettlr, both.

**Comment by DutchPete** [comment]
`Ctrl+Alt+S` in the thin or expanded mode did not display metadata, only the title of the files.
So, in the file list I right-clicked on "Copy ID" and pasted that in a plain text editor. What was pasted was the number as shown in my original post.

**Comment by nathanlesage** [comment]
Alright, thanks. So then I'm not quite sure why it doesn't work.

AAaahhh, wait a moment. Did you really link to `[[AFR_SubSah 20190910113250.md]]`? If so, try it without the file extension, because this won't work. Either ONLY the ID, OR the full filename minus the extension!

**Comment by DutchPete** [comment]
Thanks @nathanlesage, that did the trick: no file extension in the link. Why did I not think of that myself? Duhh ...

**Comment by nathanlesage** [comment]
It's in the same paragraph in the docs from where you've quoted:

> First, if the contents of the link (in the above example "This is the link") exactly matches a filename, **excluding its extension**, the appropriate file will report that it is indeed an exact match.

:D

**Comment by DutchPete** [comment]
Yes, I saw that, but I interpreted it differently: for me it meant that, although the extension is there, in fact must be there for completeness sake, Zettlr excludes it (i.e does not consider it) when it tries to find an exact match.

Perhaps it would be clearer if the text states explicitly that in the syntax of an internal link the extension **must** be left out. Adding an example always helps, e.g. `filename.md` in an internal link must be shown as `[[filename]]` and not a `[[filename.md]]`.

That way there will not be any misunderstanding.
